toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/components/formelement/AttachmentFormElement.kt

```diff
@@ -0,0 +1,458 @@
+            ),
+            overflow = TextOverflow.Ellipsis,
+            modifier = Modifier
+                .padding(horizontal = 1.dp)
+                .align(Alignment.CenterHorizontally)
+        
+        )
+        Spacer(modifier = Modifier.height(5.dp))
+    }
+}
+
+@Composable
+private fun AddAttachmentMenu(expanded: Boolean, onDismiss: () -> Unit = {}) {
+    MaterialTheme(shapes = MaterialTheme.shapes.copy(extraSmall = RoundedCornerShape(8.dp))) {
+        DropdownMenu(
+            expanded = expanded,
+            offset = DpOffset.Zero,
+            onDismissRequest = onDismiss
+        ) {
+            DropdownMenuItem(
+                text= { Text(text = "Take Photo") },
+                trailingIcon = {
+                    Icon(
+                        imageVector = Icons.Rounded.AddAPhoto,
+                        contentDescription = "Add a photo",
+                        modifier = Modifier.alpha(0.4f)
+                    )
+                },
+                onClick = {}
+            )
+            Divider(modifier = Modifier.padding(vertical = 5.dp))
+            DropdownMenuItem(
+                text= { Text(text = "Add Photo From Gallery") },
+                trailingIcon = {
+                    Icon(
+                        imageVector = Icons.Rounded.AddPhotoAlternate,
+                        contentDescription = "Add photo from gallery",
+                        modifier = Modifier.alpha(0.4f)
+                    )
+                },
+                onClick = {}
+            )
+            Divider(modifier = Modifier.padding(vertical = 5.dp))
+            DropdownMenuItem(
+                text= { Text(text = "Add File") },
+                trailingIcon = {
+                    Icon(
+                        imageVector = Icons.Rounded.LibraryAdd,
+                        contentDescription = "Add File",
+                        modifier = Modifier.alpha(0.4f)
+                    )
+                },
+                onClick = {}
+            )
+        }
+    }
+    
+}
+
+
+@Composable
+private fun AddAttachment() {
+    var showMenu by remember { mutableStateOf(false) }
+    
+    Row {
+        Box(
+            modifier = Modifier
+                .size(40.dp)
+                .feedbackClickable {
+                    showMenu = true
+                }
+        ) {
+            AddAttachmentMenu(expanded = showMenu, onDismiss = { showMenu = false })
+            Icon(
+                Icons.Rounded.Add,
+                contentDescription = "Add attachment",
+                modifier = Modifier
+                    .align(Alignment.Center)
+                    .size(32.dp)
+            )
+        }
+        
+    }
+}
+
+@Composable
+private fun AttachmentElementHeader(
+    title: String,
+    description: String,
+    modifier: Modifier = Modifier,
+    showingDetails: Boolean = false,
+    editable: Boolean = true
+) {
+    Row(
+        modifier = modifier.height(84.dp),
```

Comment:
```suggestion
        modifier = modifier.wrapContentHeight(),
```
`wrapContentHeight()` might be better here so it dynamically resizes the header height.

Reply:
now it looks correct to me. good tip 👍